src/py21cmfast/drivers/param_config.py

```diff
@@ -0,0 +1,440 @@
+#         return redshift
+
+
+# def _setup_inputs(
+#     input_params: dict[str, Any],
+#     input_boxes: dict[str, OutputStruct] | None = None,
+#     redshift=-1,
+# ):
+#     """
+#     Verify and set up input parameters to any function that runs C code.
+
+#     Parameters
+#     ----------
+#     input_boxes
+#         A dictionary of OutputStruct objects that are meant as inputs to the current
+#         calculation. These will be verified against each other, and also used to
+#         determine redshift, if appropriate.
+#     input_params
+#         A dictionary of keys and dicts / input structs. This should have the random
+#         seed, cosmo/user params and optionally the flag and astro params.
+#     redshift
+#         Optional value of the redshift. Can be None. If not provided, no redshift is
+#         returned.
+
+#     Returns
+#     -------
+#     random_seed
+#         The random seed to use, determined from either explicit input or input boxes.
+#     input_params
+#         The configured input parameter structs, in the order in which they were given.
+#     redshift
+#         If redshift is given, it will also be output.
+#     """
+#     input_boxes = input_boxes or {}
+
+#     if "flag_options" in input_params and "user_params" not in input_params:
+#         raise ValueError("To set flag_options requires user_params")
+#     if "astro_params" in input_params and "flag_options" not in input_params:
+#         raise ValueError("To set astro_params requires flag_options")
+
+#     keys = list(input_params.keys())
+#     pkeys = ["user_params", "cosmo_params", "astro_params", "flag_options"]
+
+#     # Convert the input params into the correct classes, unless they are None.
+#     outparams = convert_input_dicts(*[input_params.pop(k, None) for k in pkeys])
+
+#     # Get defaults from datasets where available
+#     params = _configure_inputs(
+#         list(zip(pkeys, outparams)) + list(input_params.items()),
+#         *list(input_boxes.values()),
+#     )
+
+#     if redshift != -1:
+#         redshift = configure_redshift(
+#             redshift,
+#             *[
+#                 v
+#                 for k, v in input_boxes.items()
+#                 if hasattr(v, "redshift") and "prev" not in k
+#             ],
+#         )
+
+#     p = convert_input_dicts(*params[:4], defaults=True)
+
+#     # This turns params into a dict with all the input parameters in it.
+#     params = dict(zip(pkeys + list(input_params.keys()), list(p) + params[4:]))
+
+#     # Sort the params back into input order and ignore params not in input_params.
+#     params = dict(zip(keys, [params[k] for k in keys]))
+
+#     # Perform validation between different sets of inputs.
+#     validate_all_inputs(**{k: v for k, v in params.items() if k != "random_seed"})
+
+#     # return as list of values
+#     params = list(params.values())
+
+#     out = params
+#     if redshift != -1:
+#         out.append(redshift)
+
+#     return out
```

Comment:
Yeah I think this can all be removed.

Reply:
Should I add a redshift check?

Reply:
hmmm not sure -- what exactly do you mean?

Reply:
As in including `random_seed` and `redshift` in all the `InputParameters` consistency checks, I'm experimenting with this now

Reply:
Ah, I don't think we need to do consistency checks for `redshift` or `random_seed`, because we never accept both those parameters _and_ output structs that have them, like we did before

Reply:
But we can still pass inconsistent output structs into the single_field functions (same parameters different seed/redshift). If you think what I've implemented in the last commit is unhelpful we can undo it.